h2o-extensions/target-encoder/src/main/java/ai/h2o/targetencoding/TargetEncoderModel.java

```diff
@@ -422,18 +472,34 @@ private void reorderColumns(Frame fr) {
     String[] toTheEnd = _parms.getNonPredictors();
     Map<String, Integer> nameToIdx = nameToIndex(fr);
     List<Integer> toAppendAfterNumericals = new ArrayList<>();
-    String[] columns = fr.names();
-    int[] newOrder = new int[columns.length];
+    String[] trainColumns = _parms.train().names();
```

Comment:
As discussed, `TargetEncoderModel` cannot have reference the training frame, otherwise SW won't be able to delete the training frame during the call of SW's `fit` method. This will cause a memory leak since a user can't access frames created inside SW's `fit` method.

Reply:
@mn-mikke this should be fixed by now